src/cli.rs

```diff
@@ -3,20 +3,15 @@ use std::io;
 pub fn get_number_of_players() -> usize {
     println!("How many players?");
     loop {
-        let number_of_players = user_input();
-
-        let number_of_players: usize = match number_of_players.trim().parse() {
-            Ok(nop) => nop,
+        match user_input().trim().parse() {
+            Ok(0) => {
+                println!("You can't play and not play at the same time. This isn't Quantum Theory.");
+            },
```

Comment:
isn't here a `continue` missing?

Reply:
By moving all the conditions into the `match`, we're able to return directly when the user inputs a string that parses to a non-zero `usize`  (on line 10). The previous use of `continue` was to avoid a `return` in the looped block. Now that we return directly, there's no `return` line to avoid and, if we hit the `Ok(0)` or `Err(_)` branches, we'll naturally continue on with the loop.